spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkConfParser.java

```diff
@@ -264,7 +261,26 @@ public ThisT tableProperty(String name) {
       return self();
     }
 
-    protected T parse(Function<String, T> conversion, T defaultValue) {
+    public ThisT check(Predicate<T> newCheck, String newErrMsg) {
+      this.check = newCheck;
+      this.errMsg = newErrMsg;
+      return self();
+    }
+
+    protected T parseAndCheck(Function<String, T> conversion, T defaultValue) {
+      T value = parse(conversion, defaultValue);
+      Preconditions.checkArgument(value != null, "Value cannot be null");
+      check(value);
+      return value;
+    }
+
+    protected T parseOptionalAndCheck(Function<String, T> conversion, T defaultValue) {
+      T value = parse(conversion, defaultValue);
```

Comment:
Why do we need a new method? In the existing parse() methods, if check != null can be the case to trigger the check?

Reply:
You mean what's the benefit of having `parseOptionalAndCheck` if we can simply call `check` in children?

Reply:
Oh I mean like adding in existing base ConfParser.parse(Function<String, T> conversion, T defaultValue) method

```
if (check != null) {
   check(value);
}
```